src/NServiceBus.Core/Transports/Development/DevelopmentTransportDispatcher.cs

```diff
@@ -0,0 +1,171 @@
+                    WriteMessage(subscriber, transportOperation, transaction);
+                }
+            }
+        }
+
+
+        void DispatchUnicast(IEnumerable<UnicastTransportOperation> transportOperations, TransportTransaction transaction)
+        {
+            foreach (var transportOperation in transportOperations)
+            {
+                WriteMessage(transportOperation.Destination, transportOperation, transaction);
+            }
+        }
+
+        void WriteMessage(string destination, IOutgoingTransportOperation transportOperation, TransportTransaction transaction)
+        {
+            var nativeMessageId = Guid.NewGuid().ToString();
+            var destinationPath = Path.Combine(basePath, destination);
+            var bodyDir = Path.Combine(destinationPath, ".bodies");
+
+            Directory.CreateDirectory(bodyDir);
+
+
+            var bodyPath = Path.Combine(bodyDir, nativeMessageId) + ".xml"; //TODO: pick the correct ending based on the serialized type
+
+            File.WriteAllBytes(bodyPath, transportOperation.Message.Body);
+
+            var messageContents = new List<string>
+            {
+                bodyPath,
+                HeaderSerializer.Serialize(transportOperation.Message.Headers)
+            };
+
+            DateTime? timeToDeliver = null;
+            DelayDeliveryWith delayDeliveryWith;
+
+            if (transportOperation.DeliveryConstraints.TryGet(out delayDeliveryWith))
+            {
+                timeToDeliver = DateTime.UtcNow + delayDeliveryWith.Delay;
+            }
+
+            DoNotDeliverBefore doNotDeliverBefore;
+
+            if (transportOperation.DeliveryConstraints.TryGet(out doNotDeliverBefore))
+            {
+                timeToDeliver = doNotDeliverBefore.At;
+            }
+
+
+            if (timeToDeliver.HasValue)
+            {
+                if (timeToDeliver.Value.Millisecond > 0)
+                {
+                    timeToDeliver += TimeSpan.FromSeconds(1);
+                }
+
+                destinationPath = Path.Combine(destinationPath, ".delayed", timeToDeliver.Value.ToString("yyyyMMddHHmmss"));
+
+                Directory.CreateDirectory(destinationPath);
+            }
+
+            var messagePath = Path.Combine(destinationPath, nativeMessageId) + ".txt";
+
+            IDevelopmentTransportTransaction directoryBasedTransaction;
+
+            if (transportOperation.RequiredDispatchConsistency != DispatchConsistency.Isolated &&
+                transaction.TryGet(out directoryBasedTransaction))
+            {
+                directoryBasedTransaction.Enlist(messagePath, messageContents);
+            }
+            else
+            {
+                var tempFile = Path.GetTempFileName();
+
+                //write to temp file first so we can do a atomic move
+                //this avoids the file being locked when the receiver tries to process it
+                File.WriteAllLines(tempFile, messageContents);
```

Comment:
Is this necessary when we use the different overloads on the receiver side for the file access and sharing?